dev-libs/optix/optix-7.3.0.ebuild

```diff
@@ -0,0 +1,124 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake cuda eapi8-dosym
+
+DESCRIPTION="NVIDIA Ray Tracing Engine"
+HOMEPAGE="https://developer.nvidia.com/optix"
+SRC_URI="NVIDIA-OptiX-SDK-${PV}-linux64-x86_64.sh"
+S="${WORKDIR}"
+
+LICENSE="OptiX-SDK"
+SLOT="0/7"
+KEYWORDS="~amd64"
+IUSE="samples"
+
+RESTRICT="fetch mirror"
+
+RDEPEND="
+	dev-util/nvidia-cuda-toolkit
+	media-libs/freeglut
+	virtual/opengl
+	samples? (
+		dev-libs/libbsd
+		media-libs/openexr:0=
+		media-libs/vulkan-loader
+		sys-libs/zlib
+		virtual/opengl
+		x11-libs/libICE
+		x11-libs/libX11
+		x11-libs/libXau
+		x11-libs/libXcursor
+		x11-libs/libXdmcp
+		x11-libs/libXext
+		x11-libs/libXfixes
+		x11-libs/libXinerama
+		x11-libs/libXrandr
+		x11-libs/libXrender
+		x11-libs/libXxf86vm
+		x11-libs/libxcb:=
+	)
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	samples? (
+		app-admin/chrpath
+		dev-libs/libbsd
+		media-libs/openexr:0
+		media-libs/vulkan-loader
+		sys-libs/zlib
+		virtual/opengl
+		x11-libs/libICE
+		x11-libs/libX11
+		x11-libs/libXau
+		x11-libs/libXcursor
+		x11-libs/libXdmcp
+		x11-libs/libXext
+		x11-libs/libXfixes
+		x11-libs/libXinerama
+		x11-libs/libXrandr
+		x11-libs/libXrender
+		x11-libs/libXxf86vm
+		x11-libs/libxcb
+	)
+"
```

Comment:
Looks like I'm still confused sometimes by the BDEPEND dependency class...
I was thinking, it needs to be on CBUILD, so it belongs to BDEPEND. And it has to go into RDEPEND to the samples can run on CHOST. Am I wrong?

Reply:
BDEPEND is essentially only for things that needs to be executed at build-time on CBUILD, like doc generation, chrpath, pkgconfig, etc... having binaries or libraries from another arch that you can't run wouldn't be useful.

If you're cross-compiling for e.g. aarch64 CHOST using an amd64 CBUILD, then you don't need the amd64 libX11 and rather only the aarch64 libX11 that'd be in DEPEND for build-time (plus its headers, which may be different than the amd64 ones, thus why xorg-proto headers package also goes in DEPEND rather than BDEPEND).